lib/SetAPI/sampleset/SampleSetInterface.py

```diff
@@ -1,65 +1,72 @@
 """An interface for handling sample sets."""
+from copy import deepcopy
+from typing import Any
 
-import traceback
-from pprint import pprint
+from installed_clients.WorkspaceClient import Workspace
 
-from SetAPI.util import convert_workspace_param, dfu_get_obj_data, info_to_ref
+from SetAPI.generic.SetInterfaceV1 import SetInterfaceV1
 
 
 class SampleSetInterface:
-    def __init__(self, workspace_client):
+    def __init__(self: "SampleSetInterface", workspace_client: Workspace):
         self.ws_client = workspace_client
-
-    def _check_condition_matching(self, conditionset_ref, matching_conditions):
-        conditionset_data = dfu_get_obj_data(conditionset_ref)
-        conditions = list(conditionset_data.get("conditions").keys())
-
-        if not all(x in conditions for x in matching_conditions):
-            error_msg = f"ERROR: Given conditions ({matching_conditions}) do not match ConditionSet conditions: {conditions}"
-            raise ValueError(error_msg)
-
-    def create_sample_set(self, ctx, params):
-        # check the ws_id param to see if it is actually a ws_name to ensure
-        # we use the right form of workspace identifier when saving
-        ws_params = convert_workspace_param({"workspace": params.get("ws_id")})
-
-        params["sample_ids"] = []
-        params["condition"] = []
+        self.set_interface = SetInterfaceV1(workspace_client)
+
+    @staticmethod
+    def set_type() -> str:
+        """The set type saved by this class."""
+        return "KBaseRNASeq.RNASeqSampleSet"
+
+    def create_sample_set(
+        self: "SampleSetInterface", ctx: dict[str, Any], params: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Save a sample set to the workspace.
+
+        :param self: this class
+        :type self: SampleSetInterface
+        :param ctx: KBase context
+        :type ctx: dict[str, Any]
+        :param params: request parameters
+        :type params: dict[str, Any]
+        :return: result of saving sampleset to the workspace
+        :rtype: dict[str, Any]
+        """
+        output_object = deepcopy(params)
+
+        # N.b. according to the spec, these are valid input params.
+        # Just hope no one is trying to use them, eh?
+        output_object["sample_ids"] = []
+        output_object["condition"] = []
```

Comment:
At some point in a future PR I'll remove those params from the spec so people aren't confused as to why their sample sets aren't getting saved.